### PR TITLE
Fix two lgtm.com alerts: potential resource leaks

### PR DESCRIPTION
**Pull request by sjvs**
This PR addresses two alerts reported by lgtm.com (@lgtmhq):

```HttpDiscovery.java```: https://lgtm.com/projects/g/bitcoinj/bitcoinj/snapshot/df1f776398d196d24e48653c8c952b97370b1e19/files/core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java#V100

```BuildCheckpoints.java```: https://lgtm.com/projects/g/bitcoinj/bitcoinj/snapshot/df1f776398d196d24e48653c8c952b97370b1e19/files/tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java#V214

There are a total of 29 alerts on lgtm.com for bitcoinj - most of them look legit: https://lgtm.com/projects/g/bitcoinj/bitcoinj/alerts/

If you enable automatic code reviews through PR integration, you can actually catch things like this before they get merged into master :smiley:!

### CODE REVIEW

**Comment by sjvs** [comment]
Had a look at the test failure: not sure how that could have been caused by my changes?

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java [hunk 1]
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
...
 
     private static void sanityCheck(File file, int expectedSize) throws IOException {
-        CheckpointManager manager = new CheckpointManager(params, new FileInputStream(file));
+    	FileInputStream fis = new FileInputStream(file);
+        CheckpointManager manager;
+
+        try {
+            manager = new CheckpointManager(params, fis);
+        } catch (IOException ioe){
+            fis.close();
```
Comment (schildbach) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: Why not put the fis.close() in a finally block?

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java [hunk 2]
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
     }
 
     private static void sanityCheck(File file, int expectedSize) throws IOException {
-        CheckpointManager manager = new CheckpointManager(params, new FileInputStream(file));
+    	FileInputStream fis = new FileInputStream(file);
```
Comment (schildbach) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: Can you squash this commit with the indenting fix you already added?

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java [hunk 3]
```diff
@@ -211,7 +211,16 @@ private static void writeTextualCheckpoints(TreeMap<Integer, StoredBlock> checkp
...
+        CheckpointManager manager;
+
+        try {
+            manager = new CheckpointManager(params, fis);
+        } catch (IOException ioe){
```
Comment (schildbach) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: We prefer a space before the opening bracket (if it's at EOL).

File: core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java [hunk 4]
```diff
@@ -97,9 +97,20 @@ public HttpDiscovery(NetworkParameters params, Details details,  OkHttpClient cl
...
-            stream.close();
+            PeerSeedProtos.SignedPeerSeeds proto;
+            try {
+                proto = PeerSeedProtos.SignedPeerSeeds.parseDelimitedFrom(zip);
+            } catch (IOException ioe){
+                zip.close(); // will close InputStream as well
+                throw ioe;
+            }
+            
+            if (zip != null){
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java]: Why the null check? Looks as if zip can't be null here.

File: core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java [hunk 5]
```diff
@@ -97,9 +97,20 @@ public HttpDiscovery(NetworkParameters params, Details details,  OkHttpClient cl
...
             InputStream stream = response.body().byteStream();
+            
             GZIPInputStream zip = new GZIPInputStream(stream);
-            PeerSeedProtos.SignedPeerSeeds proto = PeerSeedProtos.SignedPeerSeeds.parseDelimitedFrom(zip);
-            stream.close();
+            PeerSeedProtos.SignedPeerSeeds proto;
+            try {
+                proto = PeerSeedProtos.SignedPeerSeeds.parseDelimitedFrom(zip);
+            } catch (IOException ioe){
+                zip.close(); // will close InputStream as well
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/net/discovery/HttpDiscovery.java]: Again zip.close() could be in a finally block?

**Comment by schildbach** [comment]
Thanks for the PR! I added some minor comments.

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java (hunk 3 above)
Comment (sjvs) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: will fix

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java (hunk 1 above)
Comment (sjvs) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: ugh - weirdly I thought try-finally wasn't introduced until after Java 6, but that's of course try-with-resources. Ignore me. Will fix.

**Comment by sjvs** [comment]
@schildbach: I think I've addressed your comments.

I had a quick look at the other lgtm.com alerts (https://lgtm.com/projects/g/bitcoinj/bitcoinj/alerts/): a lot of them look legit and well worth fixing, but it requires someone who's a bit more confident with the bitcoinj than me (i.e., you?).

Let me know if there's anything else you'd like me to do in this PR, and keep up the good work!

**Comment by schildbach** [comment]
I just squashed your commits a bit and merged! Thanks for bearing with me and sorry for the delay.

**Pull request closed without merging**